### ISSUE TITLE
[FEATURE] Allow use of custom credentials or secrets via variables and tools

### ISSUE THREAD

**Issue by PylotLight** [FlowiseAI/Flowise#3654]
**Describe the feature you'd like**
Allow use of custom credentials or secrets via variables and tools, so this would look like either creating a "generic" item in credentials list that you could reference via the same var.Credential pattern, or set a variable to secret type, not sure which is better.

**Additional context**
I.e I have an Azure SAS token consumed via custom tool and ideally I wouldn't be putting this value exposed in a var, but credential, but don't see how I would consume the value of a credential/secret in a tool, as currently I have to put the SAS token in the plain text variables section.

**Comment by dbsanfte** [comment]
I second this, this is very important for secure usage in a corporate environment, we can't hardcode secrets.

**Comment by Rock-son** [comment]
Exactly, it is stated in "How to use" that it's possible to use double braces, but then they don't work at all:
```
{
    "url": "https://api.example.com/endpoint/sse",
    "headers": {
        "Authorization": "Bearer {{$vars.var1}}"
    }
}
```
